pennylane/debugging.py

```diff
@@ -105,3 +109,90 @@ def get_snapshots(*args, **kwargs):
         return dbg.snapshots
 
     return get_snapshots
+
+
+class PLDB(pdb.Pdb):
+    """Custom debugging class integrated with Pdb."""
+
+    __active_dev = None
+
+    def __init__(self, *args, **kwargs):
+        """Initialize the debugger, and set custom prompt string."""
+        super().__init__(*args, **kwargs)
+        self.prompt = "[pldb]: "
+
+    @classmethod
+    def valid_context(cls):
+        """Determine if the debugger is called in a valid context.
+
+        Raises:
+            RuntimeError: Can't call breakpoint outside of a qnode execution
+            TypeError: Breakpoints not supported on this device
+        """
+
+        if not qml.queuing.QueuingManager.recording() or not cls.is_active_dev():
+            raise RuntimeError("Can't call breakpoint outside of a qnode execution")
+
+        if cls.get_active_device().name not in ("default.qubit", "lightning.qubit"):
+            raise TypeError("Breakpoints not supported on this device")
+
+    @classmethod
+    def add_device(cls, dev):
+        """Update the global active device.
+
+        Args:
+            dev (Union[Device, "qml.devices.Device"]): The active device
+        """
+        cls.__active_dev = dev
+
+    @classmethod
+    def get_active_device(cls):
+        """Return the active device.
+
+        Raises:
+            RuntimeError: No active device to get
+
+        Returns:
+            Union[Device, "qml.devices.Device"]: The active device
+        """
+        if not cls.is_active_dev():
+            raise RuntimeError("No active device to get")
+
+        return cls.__active_dev
+
+    @classmethod
+    def is_active_dev(cls):
```

Comment:
```suggestion
    def has_active_dev(cls):
```

For me "is" would be a check like "is this device the active device".  But it looks more like this method is checking "does an active device exist?"

Reply:
Fixed